src/Router.tsx

```diff
@@ -0,0 +1,51 @@
+import { useContext, useEffect } from "react";
+import { Routes, Route, useSearchParams } from "react-router-dom";
+import GlobalStyles from "./components/global-styles";
+import Navbar from "./components/navbar";
+import Request from "./components/request";
+import Layout from "./components/layout";
+import NotFound from "components/not-found";
+import TempIndex from "components/temp-index";
+import ChangeNetwork from "components/change-network/ChangeNetwork";
+import { ConnectionContext } from "context/ConnectionContext";
+import { RequestClientContext } from "context/RequestClientContext";
+
+const Router = () => {
+  const { isConnected, chainId, provider, account, signer } =
+    useContext(ConnectionContext);
+  const client = useContext(RequestClientContext);
+
+  const [searchParams] = useSearchParams();
+  const wrongNetwork =
+    isConnected &&
+    searchParams.get("chainId") &&
+    Number(searchParams.get("chainId")) !== chainId;
+
+  useEffect(() => {
+    if (isConnected && account && signer && chainId) {
+      client.setUser(account, chainId, signer);
```

Comment:
Right.I will have to think on the best way to do this, as it's a little unintuitive you have to update after setting state (but this may just be something to remember).

Reply:
yes this will eventually be fixed but for now its needed to trigger state updates internally